haarpy/permutation.py

```diff
@@ -0,0 +1,239 @@
+# Copyright 2025 Polyquantique
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Permutation matrices Python interface
+"""
+
+from math import factorial, prod
+from functools import lru_cache
+from itertools import product
+from collections.abc import Sequence
+from sympy import Symbol, simplify, binomial, factor, fraction
+from haarpy import set_partitions, meet_operation, join_operation, partial_order
+
+
+@lru_cache
+def mobius_function(
+    partition_1: tuple[tuple[int]], partition_2: tuple[tuple[int]]
+) -> int:
+    """Return the Möbius function as seen in Collin & Nagatsu's
+    "Weingarten calculus for centered random permutation matrices"
```

Comment:
Should we use a standard format for papers? Should we at least add the arXiv identifier?